### PR TITLE
ECS Queue Handler sleep if limits reached

### PR DESCRIPTION
**Pull request by m2bright**

### CODE REVIEW

File: rapid/workflow/queue_handlers/handlers/ecs_queue_handler.py [hunk 1]
```diff
@@ -153,14 +158,17 @@ def _run_task(self, task_definition):
             response_dict = ecs_client.run_task(**task_definition)
             if response_dict['failures']:
                 logger.error("Failures were found: [{}]".format(response_dict['failures']))
-                status_id = StatusConstants.FAILED
+                if 'limit' in response_dict['failures'][0]['reason']:
```
Comment (jlesueur) [inline rapid/workflow/queue_handlers/handlers/ecs_queue_handler.py]: just curious, do they have a code, or a http status they use that is specific? or is there only this error message?

File: rapid/workflow/queue_handlers/handlers/ecs_queue_handler.py [hunk 2]
```diff
@@ -153,14 +158,17 @@ def _run_task(self, task_definition):
...
+                else:
+                    status_id = StatusConstants.FAILED
             elif not response_dict['tasks']:
                 logger.error("No tasks were returned")
                 status_id = StatusConstants.READY
             elif response_dict['tasks']:
                 for task in response_dict['tasks']:
                     assigned_to = '--ecs--{}'.format(task['taskArn'])
-        except (KeyError, ClientError) as exception:
+        except (KeyError, ClientError, IndexError) as exception:
```
Comment (jlesueur) [inline rapid/workflow/queue_handlers/handlers/ecs_queue_handler.py]: you might get index error if the structure of the response isn't what you expect? in which case, it's just an unknown failure, like the "else" above. 👍

File: rapid/workflow/queue_handlers/handlers/ecs_queue_handler.py (hunk 1 above)
Comment (m2bright) [reply rapid/workflow/queue_handlers/handlers/ecs_queue_handler.py]: It is only this error message, unfortunately. There are no error types nor error codes. It is quite frustrating.

File: rapid/workflow/queue_handlers/handlers/ecs_queue_handler.py (hunk 2 above)
Comment (m2bright) [reply rapid/workflow/queue_handlers/handlers/ecs_queue_handler.py]: Yep. That is right.

**Pull request merged**